mlrun/datastore/targets.py

```diff
@@ -1644,15 +1660,20 @@ def as_df(
         db_path, table_name, _, _, _, _ = self._parse_url()
         engine = sqlalchemy.create_engine(db_path)
         with engine.connect() as conn:
-            df = pd.read_sql(
-                f"SELECT * FROM {self.attributes.get('table_name')}",
-                con=conn,
-                parse_dates=self.attributes.get("time_fields"),
+            # TODO : filter as part of the query
+            df = filter_df_start_end_time(
+                pd.read_sql(
+                    f"SELECT * FROM {self.attributes.get('table_name')}",
```

Comment:
Preexisting, but important. We have a SQL injection situation here. Need to use `read_sql`'s `params` parameter instead.

Reply:
Opened a ticket, as discussed [ML-3922](https://jira.iguazeng.com/browse/ML-3922).